libs/components/datetime/src/lib/modules/date-range-picker/types/date-range-calculator.ts

```diff
@@ -10,36 +13,53 @@ import { SkyDateRangeCalculatorType } from './date-range-calculator-type';
  * Represents the calculator.
  */
 export class SkyDateRangeCalculator {
+  #config: SkyDateRangeCalculatorConfig;
+  #resourcesSvc: SkyLibResourcesService;
+
   /**
    * The text to display in the calculator select menu.
    */
-  public shortDescription: string;
+  public get shortDescription$(): Observable<string> {
+    if (this.shortDescription) {
+      return of(this.shortDescription);
+    }
+
+    if (!this.#shortDescriptionResourceKey) {
+      throw new Error(
+        'Calculator created without short description or resource key.',
+      );
+    }
+
+    return this.#resourcesSvc.getString(this.#shortDescriptionResourceKey);
+  }
 
   /**
-   * @internal
+   * The text to display in the calculator select menu.
+   * @deprecated Subscribe to the `shortDescription$` observable instead.
    */
```

Comment:
Can we update the `Custom calculator` code example showing consumers how to use the new observable?